proxy/src/proxy.rs

```diff
@@ -153,6 +154,9 @@ async fn handshake<S: AsyncRead + AsyncWrite + Unpin>(
                     creds.sni_data = tls.get_ref().1.sni_hostname().map(|s| s.to_owned());
                 }
 
+                // Set common_name
+                creds.common_name = common_name.clone();
```

Comment:
I'm not convinced that common name should be cloned. We have only one common name during the runtime, so why should we clone it instead of providing a read only reference to a single string?

Besides, the current implementation changes the `creds` object after it has been constructed, which is a code smell. We should provide a `build` or `parse` function accepting all the prerequisites instead.

Reply:
On the first note: I think I added .clone() bc the compiler was complaining. I'll check again.
On the second note: I agree. I thought it was fine because creds.sni_data was also modified. Will refactor this.

Reply:
will wait to handle this on monday